### PR TITLE
provision: make sure we don't disable persistent NIC names

### PR DESCRIPTION
**Pull request by EmilienM**
I tested it in prod and it works \o/

### CODE REVIEW

File: provision.sh [hunk 1]
```diff
@@ -275,7 +275,11 @@ if [[ $CLUSTER_NAME == *"az"* ]]; then
     $SCP_CMD $ROOT_DIR/secrets/osp-ci/exported-data/$CENTRAL_NAME $SERVER_USER@$PUBLIC_IP:/tmp/exported-data
     $SSH_CMD "bash -c 'sudo mv /tmp/exported-data /opt'"
 fi
-    
+
+# We need persistent NIC names otherwise we can have networking issues after reboots
+echo "DEBUG: Ensure that Persistent NIC naming is not disabled..."
+$SSH_CMD "if grep -q net.ifnames=0 /etc/default/grub; then sudo sed -i 's/net.ifnames=0//g' /etc/default/grub && sudo reboot; fi"
```
Comment (mdbooth) [inline provision.sh:281]: Shouldn't this live in dev-install? It's presumably something there which is setting it.

Comment (mdbooth) [reply provision.sh:281]: Incidentally, based on nothing but a hunch this change would be high on my list of suspects: https://github.com/shiftstack/dev-install/commit/3e149182d8ebeb2c9ce18b83254a25ac825c3676

FWIW, I definitely don't trust the patch author.

**Comment by mdbooth** [comment]
It's part of the CentOS 8 base cloud image!!! 🤯

```
$ qemu-img create -f qcow2 -b https://cloud.centos.org/centos/8/x86_64/images/CentOS-8-GenericCloud-8.4.2105-20210603.0.x86_64.qcow2 centos.qcow2
$ guestfish -i centos.qcow2
><fs> cat /etc/default/grub
GRUB_TIMEOUT=1
GRUB_DISTRIBUTOR="$(sed 's, release .*$,,g' /etc/system-release)"
GRUB_DEFAULT=saved
GRUB_DISABLE_SUBMENU=true
GRUB_TERMINAL_OUTPUT="console"
GRUB_CMDLINE_LINUX="console=ttyS0,115200n8 no_timer_check net.ifnames=0 crashkernel=auto"
GRUB_DISABLE_RECOVERY="true"
GRUB_ENABLE_BLSCFG=true
```

I think we should put this in dev-install along with other stupid hacks for when using CentOS.

**Comment by EmilienM** [comment]
> I think we should put this in dev-install along with other stupid hacks for when using CentOS.

ok I'll do that.

**Comment by EmilienM** [comment]
https://github.com/shiftstack/dev-install/pull/133

**Pull request closed without merging**